testfiles/joblevelpermskb/output/dependabot-run-steps.yml

```diff
@@ -0,0 +1,72 @@
+name: Dependabot       
+on:
+  push:
+  
+jobs:
+  Dependabot Pull Request:
+    permissions:
+      pull-requests: write # for dependabot/fetch-metadata to create a pr with updated dependencies
+    runs-on: ubuntu-latest
+    steps:
+    - name: Fetch Dependabot metadata
+      id: dependabot-metadata
+      uses: dependabot/fetch-metadata@v1.1.1
+      with:
+        github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+  Dependabot auto-approve:
+    permissions:
+      pull-requests: write # for dependabot/fetch-metadata to create a pr with updated dependencies
+    runs-on: ubuntu-latest
+    # Checking the actor will prevent your Action run failing on non-Dependabot PRs
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: dependabot-metadata
+        uses: dependabot/fetch-metadata@v1.1.1
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Approve a PR
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{github.event.pull_request.html_url}}
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+  Dependabot auto-merge:
+    permissions:
+      contents: write
```

Comment:
@Devils-Knight I realized we are not adding reasons when permissions get added via code. Will you be able to fix this?

Reply:
@varunsh-coder I have fixed the issue, Please review the changes.